From 41911aac46d69df6a205af59d60f23a418b0e875 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Wed, 2 Sep 2020 11:55:17 +0200
Subject: [PATCH] fix build without wchar

Check for wchar.h and use HAVE_WCHAR_H instead of hardcoding
POPT_WCHAR_HACK to fix the build with (embedded) toolchains that don't
support wchar

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Retrieved from:
https://github.com/rpm-software-management/popt/commit/41911aac46d69df6a205af59d60f23a418b0e875]
---
 configure.ac   | 2 +-
 src/popthelp.c | 5 ++---
 2 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 009ae7b..1fce6df 100644
--- a/configure.ac
+++ b/configure.ac
@@ -18,7 +18,7 @@ LT_INIT
 
 AC_SYS_LARGEFILE
 
-AC_CHECK_HEADERS([fnmatch.h glob.h langinfo.h libintl.h mcheck.h stdalign.h])
+AC_CHECK_HEADERS([fnmatch.h glob.h langinfo.h libintl.h mcheck.h stdalign.h wchar.h])
 
 # For some systems we know that we have ld_version scripts.
 # Use it then as default.
diff --git a/src/popthelp.c b/src/popthelp.c
index 9a3f489..5cbc1f4 100644
--- a/src/popthelp.c
+++ b/src/popthelp.c
@@ -15,8 +15,7 @@
 #include <sys/ioctl.h>
 #endif
 
-#define	POPT_WCHAR_HACK
-#ifdef 	POPT_WCHAR_HACK
+#ifdef HAVE_WCHAR_H
 #include <wchar.h>			/* for mbsrtowcs */
 #endif
 #include "poptint.h"
@@ -118,7 +117,7 @@ static size_t maxColumnWidth(FILE *fp)
 static inline size_t stringDisplayWidth(const char *s)
 {
     size_t n = strlen(s);
-#ifdef	POPT_WCHAR_HACK
+#ifdef HAVE_WCHAR_H
     mbstate_t t;
 
     memset ((void *)&t, 0, sizeof (t));	/* In initial state.  */
